src/test/java/org/jitsi/meet/test/MuteTest.java

```diff
@@ -246,29 +246,39 @@ private void muteParticipant1BeforeParticipant2JoinsAndScreenshare(String enable
         getParticipant1().getToolbar().clickVideoMuteButton();
 
         WebParticipant participant1 = getParticipant1();
-        WebParticipant participant2 = joinSecondParticipant(url2);
-        participant2.waitToJoinMUC();
-        participant2.waitForIceConnected();
-        participant2.waitForSendReceiveData(true, false);
 
-        // Wait for the media to switch over to the p2p connection.
-        TestUtils.waitMillis(2000);
-
-        // Check if p1 appears video muted on p2.
-        participant2.getFilmstrip().assertVideoMuteIcon(participant1, true);
-
-        // Start desktop share.
-        participant1.getToolbar().clickDesktopSharingButton();
-
-        DesktopSharingTest.testDesktopSharingInPresence(participant2, participant1, "desktop");
-        MeetUIUtils.waitForRemoteVideo(participant2.getDriver(), participant1.getEndpointId(), true);
-
-        // Stop desktop share and unmute video and check for video again.
-        participant1.getToolbar().clickDesktopSharingButton();
-
-        participant1.getToolbar().clickVideoMuteButton();
-        participant2.getFilmstrip().assertVideoMuteIcon(participant1, false);
-        MeetUIUtils.waitForRemoteVideo(participant2.getDriver(), participant1.getEndpointId(), true);
+        // Skip the test in p2p mode for Firefox since p2p is disabled in code for Firefox.
+        if (!participant1.getType().isFirefox() || enableP2p.equals("false"))
+        {
+            WebParticipant participant2 = joinSecondParticipant(url2);
```

Comment:
We need here after joinSecondParticipant - `participant2.waitToJoinMUC();`,

Reply:
But yeah ... if we are waiting for the p2p ice probably we can skip it ... if the check works with one participant ...

Reply:
yes ideally we wouldn't need it since we are waiting for ice, pushed a commit for it.